test/metabase/lib/test_metadata/graph_provider.cljc

```diff
@@ -65,4 +67,11 @@
 
   clojure.core.protocols/Datafiable
   (datafy [_this]
-    (list `->SimpleGraphMetadataProvider metadata-graph)))
+    (list `->SimpleGraphMetadataProvider metadata-graph))
+
+  #?@(:clj
+      [pretty/PrettyPrintable
+       (pretty [_this]
+         (if (identical? metadata-graph @(requiring-resolve 'metabase.lib.test-metadata/metadata))
+           'metabase.lib.test-metadata/metadata-provider
```

Comment:
Not really related but I changed `meta/metadata-provider` to print as its own symbol instead of an unreadable object like

```
#metabase.lib.test_metadata.graph_provider.GraphProvider[...]
```

to make test/REPL output a little nicer to look at and a little easier to work in.